### PR TITLE
Send result to email.

### PR DESCRIPTION
**Pull request by NguyenTanDuc**

### CODE REVIEW

**Comment by khanhkhanh** [comment]
Okay

File: app/controllers/exams_controller.rb [hunk 1]
```diff
@@ -45,6 +45,9 @@ def exam_params
   end
   
   def update_status_completed
-    @exam.update_status :completed if @exam.time_up?
+    if @exam.time_up?
+      @exam.update_status :completed
+      ResultMailer.result_exam(@exam).deliver_now
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: Use call_back

File: app/views/result_mailer/result_exam.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <th>Question</th>
+    <th>Your answer</th>
+    <th>Correct</th>
+  </tr>
+  <% @exam.results.each do |result| %>
+    <tr>
+      <td><%= result.question.content %></td>
+      <td><%= result.answer.content unless result.answer.nil? %></td>
+      <td>
+        <% if result.answer.nil? || !result.answer.correct? %>
```
Comment (dieunb) [inline app/views/result_mailer/result_exam.html.erb]:
```
+        <% (result.answer.nil? || !result.answer.correct?) ? "True" : "False" %>
```

File: app/views/result_mailer/result_exam.text.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+This is your exam's results
+Mark: <%= @exam.mark %>
+<% @exam.results.each do |result| %>
+  Question: <%= result.question.content %>, 
+  Your answer: <%= result.answer.content unless result.answer.nil? %>, 
+  Correct: 
+  <% if result.answer.nil? || !result.answer.correct? %>
```
Comment (dieunb) [inline app/views/result_mailer/result_exam.text.erb]: Same

File: app/models/exam.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ class Exam < ActiveRecord::Base
 
   before_create :create_default_status
   after_create :create_results
-  before_update :marking, if: :testing?
+  before_update :marking, :update_status_completed, if: :testing?
```
Comment (dieunb) [inline app/models/exam.rb]: Change making function name to another function name

**Pull request merged**